source/BoardingPanel.cpp

```diff
@@ -237,6 +248,8 @@ bool BoardingPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command,
 		int count = plunder[selected].Count();
 		
 		const Outfit *outfit = plunder[selected].GetOutfit();
+		int currentShares = Crew::SharesForShip(player.FlagshipPtr(), true, true);
+		profitShares += (plunder[selected].UnitValue() * currentShares * Depreciation::Full()) / (currentShares + 1000);
```

Comment:
Normally when we want to share profits, we pass the gross income into `Crew::ShareProfit` and let the Crew system handle it for us. However, in this case we're going to need a two-stage approach because we don't want the profit to actually get shared just yet. Instead, we want to calculate how much profit we _need_ to share.

One of the reasons for this is that we want to be able to call a convenient function and have the system handle everything for us. In the future, this will be super useful for making sure that all of the correct things happen; for example, morale increases in response to the profit being shared.

Currently, `Crew::ShareProfit` just returns the amount of profit that we need to share. That's because the morale system doesn't exist in yet in the master branch.

I propose that we add a function called `Crew::CalculateProfitShare` with the same arguments as `Crew::ShareProfit` and cut+paste the contents of `Crew::ShareProfit` into it. Then we make `Crew::ShareProfit` call `Crew::CalculateProfitShare` under the hood for now.

Then this code would be:

```cpp
		profitShare += Crew::CalculateProfitShare(
			player.Ships(),
			player.Flagship(),
			plunder[selected].UnitValue() * currentShares * Depreciation::Full()
		);
```